detect-nuget-inspector/detect-nuget-inspector/Inspection/Inspectors/SolutionInspector.cs

```diff
@@ -106,7 +121,31 @@ public Container GetContainer()
                         try
                         {
                             string projectRelativePath = project.Path;
-                            string projectPath = PathUtil.Combine(solutionDirectory, projectRelativePath);
+                            string projectPath = null;
+                            Boolean directoryPackagesExists = false;
+                            if (projectRelativePath.Contains("Directory.Packages.props"))
+                            {
+                                string parentPath = solutionDirectory;
+                                bool fileNotFound = true;
+                                while (fileNotFound)
+                                {
+                                    parentPath =
+                                        parentPath.Substring(0, parentPath.LastIndexOf("/"));
+                                    string checkFile = Path.Combine(parentPath, projectRelativePath);
+                                    if (parentPath.Equals("/"))
```

Comment:
Is the `"/"` referring to the root of the filesystem? If so, I wonder if we should be stopping well before that (no higher than the path provided to the NuGet inspector.

Reply:
The path provided to the Nuget Inspector itself is very complicated and it contains all the details right to the root of filesystem, there is no way Nuget Inspector could know the root of the project.

Reply:
Looks like we could utilize the Path class a bit more: https://learn.microsoft.com/en-us/dotnet/api/system.io.path?view=net-7.0

For example, the `GetDirectoryName` method returns the parent directory and `GetPathRoot` will give you values such as `C:\` and `/` depending on the platform.

Reply:
I found a way to determine the OperatingSystem https://learn.microsoft.com/en-us/dotnet/api/system.operatingsystem?view=net-8.0 for getting OS information